pages/api/token/temporary.ts

```diff
@@ -143,4 +160,114 @@ const sendTemporaryTokenByEmail = async (email: string, temporaryToken: string)
+
+    if (loginLockoutResponse.isAllowedToLogIn) return { next: true };
+    else {
+      res
+        .status(401)
+        .json(
+          buildUserNotAllowedToLogInErrorMessage(
+            loginLockoutResponse.remainingNumberOfAttemptsBeforeLockout,
+            loginLockoutResponse.numberOfSecondsBeforeLockoutExpires
+          )
+        );
+      return { next: false };
+    }
+  };
+};
+
+/**
+ * This is a middleware function that will validate the bearer token in the authorization header
+ *
+ * @param handler - the function to be executed next in the API call
+ * @returns either the handler to be executed next in the API call, or updates the res status and returns void
+ */
+export const hasValidFormAccessToken = (): MiddlewareRequest => {
+  return async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+    props?: MiddlewareProps
+  ): Promise<MiddlewareReturn> => {
+    assert(props?.email); // Middleware function `requestContainsEmail` ensures that email is defined
+
+    if (!process.env.TOKEN_SECRET) {
+      res.status(500).json({ error: "Server cannot respond at this time" });
+      return { next: false };
+    }
+
+    try {
+      const token = extractBearerTokenFromReq(req);
+      const { formID } = jwt.verify(token, process.env.TOKEN_SECRET) as BearerTokenPayload;
+      const tokenID = await getTokenById(formID);
+
+      if (tokenID?.bearerToken === token) return { next: true, props: { formID } };
+      else throw new Error("Invalid bearer token.");
+    } catch (err) {
+      if (err instanceof TokenExpiredError) {
+        logMessage.warn("An expired bearer token has been used.");
+      }
+
+      const loginLockoutResponse = await registerFailedLoginAttempt(props.email);
+      res
+        .status(401)
+        .json(
+          buildUserNotAllowedToLogInErrorMessage(
+            loginLockoutResponse.remainingNumberOfAttemptsBeforeLockout,
+            loginLockoutResponse.numberOfSecondsBeforeLockoutExpires
+          )
+        );
+      return { next: false };
+    }
+  };
+};
+
+function buildUserNotAllowedToLogInErrorMessage(
+  remainingNumberOfAttemptsBeforeLockout?: number,
+  numberOfSecondsBeforeLockoutExpires?: number
+) {
+  return {
+    error: "User is not allowed to log in",
+    remainingNumberOfAttemptsBeforeLockout,
+    numberOfSecondsBeforeLockoutExpires,
+  };
+}
+
+/**
+ * API outcomes:
+ * 200: `success` if everything goes well
+ * 400: `Invalid payload` if no email in request payload
+ * 401: `User is not allowed to log in` if email and/or form access token is not valid. Response paylaod also optionally contains `remainingNumberOfAttemptsBeforeLockout` (if user can still try to log in) and `numberOfSecondsBeforeLockoutExpires` (if user is locked out)
+ * 500: `Internal error` if something goes wrong while sending temporary token ; or ; `Server cannot respond at this time` if TOKEN_SECRET environment variable is not defined
+ */
+
+export default middleware(
```

Comment:
By consolidating some of the functionality above we could shorten the middleware list to something like:
```js
export default middleware(
  [
    cors({ allowedMethods: ["POST"] }),
    isUserLockedOut(),
    validBearerToken(),
  ],
  handler
);
```